### PR TITLE
Fix database password escaping

### LINKED ISSUE #2123: Database smoke tests failing on Tier 3 SRE
## :white_check_mark: Checklist

- [x] I have searched open and closed issues for duplicates.
- [x] This is a problem observed when **deploying** a Data Safe Haven.
- [x] I can reproduce this with the [latest version](https://github.com/alan-turing-institute/data-safe-haven/releases).
- [x] I have read through the [documentation](https://alan-turing-institute.github.io/data-safe-haven/).
- [x] This isn't an open-ended question (open a [discussion](https://github.com/alan-turing-institute/data-safe-haven/discussions) if it is).

## :computer: System information

- **Operating System:** macOS
- **Data Safe Haven version:** `update-nexus-allowlist` [3035d81](https://github.com/craddm/data-safe-haven/commit/3035d811abd00649ed90758f3861c30b76efa49a) (parity with `release-v5.0.0`)

## :package: Packages

<details>
<summary>List of packages</summary>

```none
Paste list of packages here
```

</details>

## :no_entry_sign: Describe the problem

Smoke tests involving databases are failing for both Python and R

## :deciduous_tree: Log messages

<details>
<summary>Relevant log messages</summary>

```none
 root@shm-green-sre-noteasybeing-vm-workspace-01:/usr/local/smoke_tests# bats run_all_tests.bats -f "database"
 ✗ MS SQL database (Python)
   (in test file run_all_tests.bats, line 101)
     `[ "$status" -eq 0 ]' failed
 ✗ MS SQL database (R)
   (in test file run_all_tests.bats, line 111)
     `[ "$status" -eq 0 ]' failed
 ✗ Postgres database (Python)
   (in test file run_all_tests.bats, line 119)
     `[ "$status" -eq 0 ]' failed
```

</details>

### PR DESCRIPTION
**Pull request by jemrobinson**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).

### :vertical_traffic_light: Depends on

n/a

### :arrow_heading_up: Summary

Fix incorrect mustache expansion of database password as this might contain HTML entities

### :closed_umbrella: Related issues

Closes #2123

### :microscope: Tests

n/a

### CODE REVIEW

File: data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ write_files:
   - path: "/etc/database_credential"
     permissions: "0400"
     content: |
-      {{ database_service_admin_password }}
+      {{{ database_service_admin_password }}}
```
Comment (JimMadge) [inline data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml:12]: This is a mustache thing?

**Review by JimMadge — APPROVED** [review]

**Comment by JimMadge** [comment]
@jemrobinson has this been tested?

**Comment by craddm** [comment]
I'm testing it now

File: data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml (hunk 1 above)
Comment (jemrobinson) [reply data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml:12]: Yes, see the "variables" section here https://mustache.github.io/mustache.5.html

**Comment by JimMadge** [comment]
@craddm Did it work for you?

**Comment by craddm** [comment]
Confirmed working

**Pull request merged**